lib/olive_branch/middleware.rb

```diff
@@ -31,5 +31,18 @@ def call(env)
         end
       end
     end
+
+    def underscore_params(env)
+      if Rails::VERSION::MAJOR < 5
```

Comment:
I'm having the same issues this patch is attempting to solve, but I'm on Rails 4.
What if we checked something else instead of the Rails version, like the existence of `env["action_dispatch.request.request_parameters"]` ?